tests/test_resolver.py

```diff
@@ -26,3 +33,145 @@ def test_open_stage_and_logging(capfd):
+        "resources/integration_test_data/recursive_assetized_resolve_to_search_path/search_path_root"
+    )
+
+    stage = open_stage(
+        "resources/integration_test_data/recursive_assetized_resolve_to_search_path/parking_lot.usd",
+        context,
+    )
+
+    assert_parking_lot_structure(stage)
+
+
+# Given a USD document that references a second level document via an
+# assetized reference resolvable by OpenAssetIO, and that second level
+# document containing a non-assetized, adjacent relative file path
+# reference to a third level document, then the document can be fully
+# resolved.
+@pytest.mark.xfail(reason="OpenAssetIO not integrated yet")
+def test_assetized_child_ref_non_assetized_grandchild():
+    stage = open_stage(
+        "resources/integration_test_data/assetized_child_ref_non_assetized_grandchild/parking_lot.usd"
+    )
+
+    assert_parking_lot_structure(stage)
+
+
+# Given a USD document that references a second level document via an
+# non-assetized, adjacent relative file path reference, and that second
+# level document containing an assetized reference resolvable by
+# OpenAssetIO to a third level document, then the document can be fully
+# resolved.
+@pytest.mark.xfail(reason="OpenAssetIO not integrated yet")
+def test_non_assetized_child_ref_assetized_grandchild():
+    stage = open_stage(
+        "resources/integration_test_data/non_assetized_child_ref_assetized_grandchild/parking_lot.usd"
+    )
+
+    assert_parking_lot_structure(stage)
+
+
+##### Utility Functions #####
+
+# Verify OpenAssetIO configured as the AR resolver.
+@pytest.fixture(autouse=True)
+def openassetio_configured():
+    pr = Plug.Registry()
+    plugin = pr.GetPluginWithName("usdOpenAssetIOResolver")
+
+    assert (
+        plugin is not None
+    ), "The usdOpenAssetIOResolver plugin is not loaded, please check the PXR_PLUGINPATH_NAME environment variable"
+
+
+# As all the data tends to follow the same form, convenience method
+# to avoid repeating myself
+def assert_parking_lot_structure(usd_stage):
+    floor = usd_stage.GetPrimAtPath("/ParkingLot/ParkingLot_Floor_1")
+
+    assert floor.IsValid()
+
+    car1 = floor.GetChild("Car1")
+    car2 = floor.GetChild("Car2")
+
+    assert car1.IsValid()
+    assert car2.IsValid()
+
+    assert car1.GetPropertyNames() == ["color"]
+    assert car2.GetPropertyNames() == ["color"]
+
+
+# Set the default resolver search path to a subdirectory
+# Done this way so that you can run the test from any directory,
+# otherwise the working directory will impact the file loading.
+def build_search_path_context(path_relative_from_file):
+    script_dir = os.path.realpath(os.path.dirname(__file__))
+    full_path = os.path.join(script_dir, path_relative_from_file)
+    return Ar.DefaultResolverContext([full_path])
+
+
+# Open the stage
+# Done this way so that you can run the test from any directory,
+# otherwise the working directory will impact the file loading.
```

Comment:
As an aside (we can tweak later), in Python, we'd normally write these as docstrings, so theyre available at runtime, and you dont need to mollify the linter.

Reply:
+1 to this

Reply:
Whilst this would be trivial, will do later anyhow.